compiler/rustc_infer/src/infer/lexical_region_resolve/mod.rs

```diff
@@ -869,7 +867,7 @@ impl<'tcx> LexicalRegionResolutions<'tcx> {
         T: TypeFoldable<'tcx>,
     {
         tcx.fold_regions(value, |r, _db| match *r {
-            ty::ReVar(rid) => self.resolve_var(rid),
+            ty::ReVar(_) => self.resolve_region(tcx, r),
             _ => r,
         })
```

Comment:
This match might be unnecessary. Could it just unconditionally try to resolve it?

Reply:
Yep, done!